### PR TITLE
migrate the code of prow tide

### PR DESCRIPTION
**Pull request by xwz123**

### CODE REVIEW

File: robot.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	return nil, fmt.Errorf("no config for this repo:%s/%s", org, repo)
+}
+
+func (bot *robot) RegisterEventHandler(f framework.HandlerRegitster) {
+	f.RegisterPullRequestHandler(bot.handlePREvent)
+	f.RegisterNoteEventHandler(bot.handleNoteEvent)
+}
+
+func (bot *robot) handlePREvent(e *sdk.PullRequestEvent, c config.Config, log *logrus.Entry) error {
+	if e.GetState() != sdk.ActionOpen {
```
Comment (zengchen1024) [inline robot.go:62]: e.GetPullRequest().GetState()

File: robot.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	ListPROperationLogs(org, repo string, number int32) ([]sdk.OperateLog, error)
+	GetBot() (sdk.User, error)
+}
+
+func newRobot(cli iClient) *robot {
+	return &robot{cli: cli}
+}
+
+type robot struct {
+	cli iClient
```
Comment (zengchen1024) [inline robot.go:36]: add a member of botName to avoid get it on pr event

File: robot.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	return bot.cli.CreatePRComment(org, repo, number, c)
+}
+
+func (bot *robot) deleteOldComments(org, repo string, number int32) error {
+	comments, err := bot.cli.ListPRComments(org, repo, number)
+	if err != nil {
+		return err
+	}
+
+	b, err := bot.cli.GetBot()
```
Comment (zengchen1024) [inline robot.go:144]: use bot.botName

File: robot.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	if !e.IsCreatingCommentEvent() || !e.IsPullRequest() {
+		return nil
+	}
+
+	if !checkPRRe.MatchString(e.GetComment().GetBody()) {
+		return nil
+	}
+
+	if !e.GetPullRequest().GetMergeable() {
+		return nil
```
Comment (zengchen1024) [inline robot.go:89]: https://github.com/opensourceways/test-infra/blob/sync-5-22/prow/gitee-plugins/tide/tide.go#L138

File: robot.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+	sdk "github.com/opensourceways/go-gitee/gitee"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const botName = "tide"
+
+var (
+	checkPRRe          = regexp.MustCompile(`(?mi)^/check-pr\s*$`)
+	tideNotification   = "@%s, This pr is not mergeable."
```
Comment (zengchen1024) [inline robot.go:18]: not used

**Review by zengchen1024 — CHANGES_REQUESTED** [review]

**Comment by zengchen1024** [comment]
think about the operation logs.

**Comment by zengchen1024** [comment]
did you reset the number of reviewer and tester to zero?

File: label.go [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+			}, true
+		}
+	}
+
+	return labelLog{}, false
+}
+
+func areAllLabelsReady(labels sets.String, cfg *botConfig) bool {
+	for _, l := range cfg.Labels {
+		if labels.Has(l.Label) {
```
Comment (chenzeng765) [inline label.go:57]: if !labels.Has(l.Label)

File: robot.go [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+	return nil, fmt.Errorf("no config for this repo:%s/%s", org, repo)
+}
+
+func (bot *robot) RegisterEventHandler(f framework.HandlerRegitster) {
+	f.RegisterPullRequestHandler(bot.handlePREvent)
+	f.RegisterNoteEventHandler(bot.handleNoteEvent)
+}
+
+func (bot *robot) handlePREvent(e *sdk.PullRequestEvent, c config.Config, log *logrus.Entry) error {
+	if e.GetPullRequest().GetState() != sdk.ActionOpen {
```
Comment (chenzeng765) [inline robot.go:64]: sdk.StatusOpen

**Review by chenzeng765 — CHANGES_REQUESTED** [review]

**Pull request merged**